src/datasets/features/audio.py

```diff
@@ -23,24 +33,51 @@ class Audio:
     _type: str = field(default="Audio", init=False, repr=False)
 
     def __call__(self):
-        return pa.string()
+        return pa.struct({"path": pa.string(), "bytes": pa.binary()})
+
+    def encode_example(self, value):
+        """Encode example into a format for Arrow.
+
+        Args:
+            value (:obj:`str` or :obj:`dict`): Data passed as input to Audio feature.
+
+        Returns:
+            :obj:`dict`
+        """
+        if isinstance(value, str):
+            return {"path": value}
+        return value
 
     def decode_example(self, value):
         """Decode example audio file into audio data.
 
         Args:
-            value: Audio file path.
+            value (obj:`str` or :obj:`dict`): Either a string with the absolute audio file path or a dictionary with
+                keys:
+
+                - path: String with absolute or relative audio file path.
+                - bytes: Optionally, the bytes of the audio file.
 
         Returns:
             dict
         """
-        # TODO: backard compatibility for users without audio dependencies
-        array, sampling_rate = (
-            self._decode_example_with_torchaudio(value)
-            if value.endswith(".mp3")
-            else self._decode_example_with_librosa(value)
-        )
-        return {"path": value, "array": array, "sampling_rate": sampling_rate}
+        if isinstance(value, str):
+            value = {"path": value, "bytes": None}
+        if value["bytes"]:
+            path, file = value["path"], BytesIO(value["bytes"])
+            array, sampling_rate = (
+                self._decode_example_with_torchaudio(file)
+                if path.endswith(".mp3")
+                else self._decode_example_with_soundfile(file)
+            )
+        else:
+            path = value["path"]
+            array, sampling_rate = (
+                self._decode_example_with_torchaudio(path)
+                if path.endswith(".mp3")
+                else self._decode_example_with_librosa(path)
+            )
+        return {"path": path, "array": array, "sampling_rate": sampling_rate}
 
     def _decode_example_with_librosa(self, value):
```

Comment:
@lhoestq I think I should rename the functions because I see they are misleading! 😅 

Before the addition of the bytes field, the audio was specified only by a path. I created 2 functions:
- For non-MP3 audio files: `_decode_example_with_librosa`
- For MP3 audio files: `_decode_example_with_torchaudio`

With the possibility of specifying the audio with a bytes field (file-object):
- For non-MP3 audio files:
  - Note that `librosa` does NOT accept file-like objects as input; I copied the `librosa` implementation (under the hood it uses `soundfile`, which accepts file-like objects), but adapting it to file-like objects; I named it `_decode_example_with_soundfile`
- For MP3 files: this is still pending to do, because I cannot test it locally (implementation with torchaudio requires `sox`, available only for Linux and my machine is Windows); that is why I implemented the Test Audio check in GitHub).

Therefore:
- For non-MP3:
  - If path, use `_decode_example_with_librosa`
  - If bytes, use `_decode_example_with_soundfile`
- For MP3:
  - If path, use `_decode_example_with_torchaudio`
  - If bytes, TODO